### ISSUE TITLE
Does cache2 cache the POST response by design?

### ISSUE THREAD

**Issue by xied75** [unbit/uwsgi#1047]
Dear all,

I have a simple flask app under uwsgi and nginx. At the beginning I was returning a 200 for a POST, surprisingly the response is cached under the same URI (e.g. /key/guid) I setup in uwsgi for GET caching. So the following GET never get anything meaningful.

Then I changed the status code to 201, and the caching stopped.

I wonder if this is documented somewhere or this is actually by design. I did some search trying to understand what is the industry norm and found this:
http://stackoverflow.com/questions/626057/is-it-possible-to-cache-post-methods-in-http
(seems there isn't a common understanding yet.)

I tried to find relevant info from the source, but failed to find anything specific handling POST (or POST response), haven't coded in C for 3 years now. :) Would be great if you could pointing me to the correct place to look.

Best regards,

Dong

**Comment by xrmx** [comment]
Thanks for reporting. I don't understand if you are already using the caching framework to cache responses like here http://uwsgi-docs.readthedocs.org/en/latest/tutorials/CachingCookbook.html or something that looks like the client caching requests. Could you please share your uwsgi config?

**Comment by xied75** [comment]
@xrmx Yes I'm using the cache2 as the link showcased. Here is the config:

```
[uwsgi]
cache2 = name=my-cache,items=10000,blocksize=6000,keysize=60
route = ^/key/ cache:key=${REQUEST_URI},name=my-cache
route = ^/key/ cachestore:key=${REQUEST_URI},name=my-cache

stats = /tmp/stats.socket
socket=/tmp/uwsgi.sock
chmod-socket=664
uid = www-data
gid = www-data

chdir=/usr/src/app/myservice
module=controller:app

master = true
processes = 4

vacuum
```

I don't think I did anything fancy above. To repeat the problem, it's the RESPONSE of my POST got cached by cache2, and after I change my controller to return 201 other than 200, the caching stopped.

I was doing my test by running uwsgicachetop for monitoring (https://github.com/goir/uwsgicachetop), I can clearly see one item got cached after I simply fire a POST via curl, when my controller returns 200; and the caching didn't happen when the code was changed to 201.

Thanks for looking into this.
